src/GraphQL/Execution/ExecutionNode.cs

```diff
@@ -35,12 +37,24 @@ public object Source
             set => _source = value;
         }
 
+        //todo: add parameter to constructor for resolvedType
         protected ExecutionNode(ExecutionNode parent, IGraphType graphType, Field field, FieldType fieldDefinition, int? indexInParentNode)
```

Comment:
Adding another parameter is a breaking change, so I suggest we add the parameter in v4.  Of course we could add another constructor to provide an optimized path when processing array nodes in v3.